yapf/__init__.py

```diff
@@ -210,13 +212,13 @@ def FormatFiles(filenames,
           print_diff=print_diff,
           verify=verify,
           logger=logging.warning)
-      if has_change and reformatted_code is not None:
-        file_resources.WriteReformattedCode(filename, reformatted_code,
-                                            in_place, encoding)
       changed |= has_change
     except SyntaxError as e:
       e.filename = filename
       raise
+    if reformatted_code is not None:
+      file_resources.WriteReformattedCode(filename, reformatted_code, in_place,
+                                          encoding)
```

Comment:
I believe the situation this was meant to solve was not writing to a file if there were no changes. Doing so updated the timestamp. It doesn't look like this is needed, though. So go ahead and move it back up please. :-)